### ISSUE TITLE
Scriptable trigger for GUI Mode

### ISSUE THREAD

**Issue by skoblenick** [hashicorp/vagrant#1849]
I would like to see the following feature added to Vagrant.

I have a box with provision scripts. I would like the box to provision without a gui until the provision scripts complete. Once the provisioning is complete it would be nice to have a scriptable trigger to alter the gui mode to be true (without reboot or halt if possible).

Presently, if I enable the gui mode and launch the box I can log into the box and modify or change things while the provision tools is executing which can cause unforeseen issues. I would like to minimize the usage of the box until it has been provisioned correctly.
